### PR TITLE
2467 Add support to save checkpoint with smaller metric

### LINKED ISSUE #2467: Inflexible score function of CheckpointSaver
https://github.com/Project-MONAI/MONAI/blob/9f9b63a7dc56d8b31d4fc8d34aae77865ba95947/monai/handlers/checkpoint_saver.py#L148-L160

Currently the `CheckpointSaver` saves the check point that has the largest score value. However, this is not desirable when a lower score indicates better performance, e.g. validation loss. We can either allow the user to pass a custom score function (as in the ignite.handlers), or add a flag to control returning `round(engine.state.metrics[metric_name], 4)` or ` - round(engine.state.metrics[metric_name], 4)`

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2467 .

### Description
This PR added support to save checkpoint with smaller metric value, because some error-like metric is `smaller is better`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/engines/evaluator.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ class Evaluator(Workflow):
             engine.state.metrics when epoch completed. key_val_metric is the main metric to compare and save the
             checkpoint into files.
         additional_metrics: more Ignite metrics that also attach to Ignite Engine.
+        metric_cmp_fn: function to compare current key metric with previous best key metric value.
+            it must accept 2 args (current_metric, previous_best) and return a bool result. default to `greater than`.
```
Comment (vfdev-5) [inline monai/engines/evaluator.py:57]:
```suggestion
            It must accept 2 args (current_metric, previous_best) and return a bool result. Default to `greater than`.
```
Same for others below. 

 I think the docstring is missing the explanation on what to return and how it will taken into account by the code. For example, if returns True, what will happen.

**Comment by Nic-Ma** [comment]
integration test `tests/test_integration_workflows.py` passed locally.

Thanks.

File: monai/engines/evaluator.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/engines/evaluator.py]: Hi @vfdev-5 , thanks for your review! I updated according to your comments.

File: monai/handlers/checkpoint_saver.py [hunk 2]
```diff
@@ -62,6 +62,8 @@ class CheckpointSaver:
             typically, it's used to resume training and compare current metric with previous N values.
         key_metric_greater_or_equal: if `True`, the latest equally scored model is stored. Otherwise,
             save the the first equally scored model. default to `False`.
+        key_metric_score_sign: sign of the score: `1.0` or `-1.0`. For error-like metrics, e.g. smaller is better,
```
Comment (wyli) [inline monai/handlers/checkpoint_saver.py:65]: This could be an int or Boolean flag

**Review by wyli — APPROVED** [review]

File: monai/handlers/checkpoint_saver.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/handlers/checkpoint_saver.py]: Thanks for your review, updated doc-string in the latest commit.

File: monai/handlers/checkpoint_saver.py [hunk 3]
```diff
@@ -62,6 +62,9 @@ class CheckpointSaver:
             typically, it's used to resume training and compare current metric with previous N values.
         key_metric_greater_or_equal: if `True`, the latest equally scored model is stored. Otherwise,
             save the the first equally scored model. default to `False`.
+        key_metric_score_sign: sign of the score, available value: `[1.0, -1.0, 1, -1]`. For error-like metrics,
```
Comment (wyli) [inline monai/handlers/checkpoint_saver.py:65]: I think this should be a flag indicating whether to flip the sign of the metric value, if it is true or positive, the sign of the metric value won't be flipped, if it's false or negative, the metric value will be flipped. It's not about supporting 1.0 -1.0 1 -1 here.

Comment (Nic-Ma) [reply monai/handlers/checkpoint_saver.py:65]: Hi @wyli ,

Thanks for your explanation. I tried to align with ignite API: https://github.com/pytorch/ignite/blob/master/ignite/handlers/checkpoint.py#L585
I am OK with both solutions, your idea seems more robust, no need to check the arg value.

Thanks.

Comment (Nic-Ma) [reply monai/handlers/checkpoint_saver.py:65]: Let me tried to update it to bool flag in this ignite related PR together: https://github.com/Project-MONAI/MONAI/pull/2492

Thanks.

Comment (wyli) [reply monai/handlers/checkpoint_saver.py:65]: That docstring seems to suggest that it'll ensure a positive or negative value, but the actual implemention is about 'flip or not'. Perhaps the ignite docstring needs an update as well...

Comment (Nic-Ma) [reply monai/handlers/checkpoint_saver.py:65]: Hi @vfdev-5 , what do you think about this arg? Maybe ignite can also change it to be a bool flag?

Thanks.

Comment (Nic-Ma) [reply monai/handlers/checkpoint_saver.py:65]: @wyli , updated in the latest commit.
Thanks.

Comment (vfdev-5) [reply monai/handlers/checkpoint_saver.py:65]: > what do you think about this arg? Maybe ignite can also change it to be a bool flag?

Hi @Nic-Ma , this is a good question. Let me see with the team about that.
